src/main/java/com/robertx22/mine_and_slash/a_libraries/dmg_number_particle/particle/InteractionResultHandler.java

```diff
@@ -20,27 +20,43 @@
 import java.util.Map;
 import java.util.Random;
 import java.util.function.BiConsumer;
+import java.util.function.Supplier;
 
 public class InteractionResultHandler {
 
 
 
     public enum ParticleSpawnType {
-        DAMAGE(ClientConfigs.getConfig().DAMAGE_PARTICLE_STYLE.get().damageStrategy, new ElementDamageParticle.DamageInformation(null, null, false)),
-        NULLIFIED_DAMAGE(ClientConfigs.getConfig().DAMAGE_PARTICLE_STYLE.get().nullifiedDamageStrategy, DamageNullifiedParticle.Type.DODGE),
-        HEAL(ClientConfigs.getConfig().DAMAGE_PARTICLE_STYLE.get().healStrategy, new HealParticle.HealNumber(0.0f));
-        public final BiConsumer<IParticleSpawnMaterial, Entity> strategy;
+        DAMAGE(new IParticleSpawnMaterial.DamageInformation(null, null, false)) {
+            @Override
+            public Supplier<BiConsumer<IParticleSpawnMaterial, Entity>> getStrategy() {
+                return () -> ClientConfigs.getConfig().DAMAGE_PARTICLE_STYLE.get().damageStrategy;
+            }
+        },
+        NULLIFIED_DAMAGE(IParticleSpawnMaterial.Type.DODGE) {
+            @Override
+            public Supplier<BiConsumer<IParticleSpawnMaterial, Entity>> getStrategy() {
+                return () -> ClientConfigs.getConfig().DAMAGE_PARTICLE_STYLE.get().nullifiedDamageStrategy;
```

Comment:
Uh what
Just return DAMAGE_PARTICLE_STYLE and grab the varible needed elsewhere

Reply:
It doesn't need 3 overrides that do mostly the same thing, and it doesn't need a supplier if it's already a method

Reply:
The ParticleSpawnType exists on both sides, but BiConsumer only exists on the client, which is why it caused a crash in the last build. 
The current code structure is that the server sends the particle material to the client. The material includes a ParticleSpawnType and can be obtained on the client side. From the ParticleSpawnType, client could know which consumer should be used for the material.

Reply:
I don't get it. Just return IParticleSpawnMaterial in the method and only call the method on the client?

Reply:
Also i generally try to never use biconsumers and stuff  unless it's super clear what's happening. A supplier<biconsumer makes me go crazy

Reply:
I move the strategy selection into the IParticleSpawnMaterial, I guess this is better than a horrible supplier of BiConsumer